core/src/services/routing.js

```diff
@@ -264,3 +265,32 @@ export const handleRouteClick = node => {
     navigateTo(route);
   }
 };
+
+const showPageNotFoundError = async (
+  component,
+  pathToRedirect,
+  notFoundPath,
+  isAnyPathMatched = false
+) => {
+  const pageNotFoundHandler = LuigiConfig.getConfigValue(
+    'routing.pageNotFoundHandler'
+  );
+
+  if (typeof pageNotFoundHandler === 'function') {
+    //custom 404 handler is provided, use it
+    pageNotFoundHandler(notFoundPath, isAnyPathMatched);
+    return;
+  }
+
+  const alert = {
+    message: isAnyPathMatched
+      ? 'Could not map the exact target node for the requested route'
+      : 'Could not find the requested route',
+    link: notFoundPath,
+    ttl: pathToRedirect === '/' ? 2 : 1 //how many redirections the alert will 'survive'.
+    //Redirecting to '/' causes another redirection to '/overview', that's why TTL needs to be bigger
```

Comment:
In our case it's ok, because we don't have a view for '/', so it redirects to '/overview', but what if someone has a view on '/' and doesn't have this additional redirection?

Reply:
🤔